analysis/trigger_performance/01_readiness.md

```diff
@@ -0,0 +1,400 @@
+    (df_readiness["Should have activated"] == "Y")
+    & (df_readiness["Readiness reached"] == "Y")
+    & (
+        df_readiness["Activation scenario"]
+        == df_readiness["Readiness scenario"]
+    ),
+    "Confusion matrix",
+] = "TP"
+# Dataset has no FPs so skipping those
+```
+
+```python
+# Rename confusion matrix column
+df_readiness = df_readiness[
+    ["International name", "Year", "Confusion matrix", "Should have activated"]
+].rename(
+    columns={
+        "International name": "typhoon",
+        "Year": "year",
+        "Confusion matrix": "readiness",
+    }
+)
+# Capitlize typhoon names
+df_readiness["typhoon"] = (
+    df_readiness["typhoon"].str.upper().str.strip().str.replace(" ", "-")
+)
+
+df_readiness
+```
+
+## Activation trigger
+
+```python
+model_results_dir = trigger_dir / "model_run_results"
+results = pd.DataFrame()
+for filename in model_results_dir.glob("*CERF_TRIGGER_LEVEL.csv"):
+    year = str(filename.name).split("_")[1][:4]
+    results["year"] = year
+    results = pd.concat([results, pd.read_csv(filename)], ignore_index=True)
+results
+```
+
+```python
+# Check if it triggered
+results["Activation reached"] = (
+    (results["80k"] >= 0.5)
+    | (results["50k"] >= 0.6)
+    | (results["30k"] >= 0.7)
+    | (results["10k"] >= 0.8)
+    | (results["5k"] >= 0.95)
+)
+results
+```
+
+```python
+# Join with readinessas
+df_activation = pd.merge(
+    left=df_readiness.rename(columns={"International name": "typhoon"}),
+    right=results[["Typhoon_name", "Activation reached"]].rename(
+        columns={"Typhoon_name": "typhoon"}
+    ),
+    how="left",
+    on="typhoon",
+)
+```
+
+```python
+# count activations
+df_activation.loc[
+    (df_activation["Should have activated"] == "Y")
+    & df_activation["Activation reached"],
+    "activation",
+] = "TP"
+
+df_activation.loc[
+    (df_activation["Should have activated"] == "Y")
+    & (df_activation["Activation reached"] == False),
+    "activation",
+] = "FN"
+
+# Here assuming that there are no false positives -- pretty safe assumption for now
+# (but I will go back and run on more historical typhoons)
+df_activation.loc[
+    (df_activation["Should have activated"] == "N")
+    & (
+        (df_activation["Activation reached"] == False)
+        # Remove this after running on more
+        | (df_activation["Activation reached"].isna())
+    ),
+    "activation",
+] = "TN"
+df_activation
+```
+
+```python
+# Final DF - compute performance of full framework
+df = df_activation.copy()
+
+df.loc[
+    (df["readiness"] == "TP") & (df["activation"] == "TP"),
+    "framework",
+] = "TP"
+
+df.loc[
+    (df["readiness"] == "TN") & (df["activation"] == "TN"),
+    "framework",
+] = "TN"
+
+df.loc[
+    ((df["readiness"] == "FN") | (df["activation"] == "FN"))
+    & (~df["activation"].isna()),
+    "framework",
+] = "FN"
+
+# Again, ignoring FPs
+df
+```
+
+```python
+# Drop unneeded columns for BSRS
+trigger_names = ["readiness", "activation", "framework"]
```

Comment:
How do you define "framework"? I see how it's computed but I wonder if we can find a clear description and contrast it with how it's defined in slow-onset frameworks. They're somewhat different, aren't they?